git_machete/client.py

```diff
@@ -1855,68 +1855,79 @@ def check_that_forkpoint_is_ancestor_or_equal_to_tip_of_branch(
+                    remote_already_added: Optional[str] = self.__get_added_remote_name_or_none(pr.repository_url)
+                    if not remote_already_added:
+                        remote_from_pr: str = pr.full_repository_name.split('/')[0]
+                        if remote_from_pr not in self.__git.get_remotes():
+                            self.__git.add_remote(remote_from_pr, pr.repository_url)
+                        remote_to_fetch: str = remote_from_pr
+                    else:
+                        remote_to_fetch = remote_already_added
+                    if remote != remote_to_fetch:
+                        print(f"Fetching {remote_to_fetch}...")
+                        self.__git.fetch_remote(remote_to_fetch)
+                    if '/'.join([remote_to_fetch, pr.head]) not in self.__git.get_remote_branches():
+                        raise MacheteException(f"Could not check out PR #{pr.number} because its head branch `{pr.head}` is already deleted from `{remote_to_fetch}`.")
+            else:
+                warn(f'Pull request #{pr.number} comes from fork and its repository is already deleted. No remote tracking data will be set up for `{pr.head}` branch.')
+                print(fmt(f"Checking out `{pr.head}` locally..."))
+                checkout_pr_refs(self.__git, remote, pr.number, LocalBranchShortName.of(pr.head))
+                self.flush_caches()
+            if pr.state == 'closed':
+                warn(f'Pull request #{pr.number} is already closed.')
+            debug('checkout_github_pr()', f'found {pr}')
+
+            path: List[LocalBranchShortName] = self.__get_path_from_pr_chain(pr, all_open_prs)
+            reversed_path: List[LocalBranchShortName] = path[::-1]  # need to add from root downwards
+            for index, branch in enumerate(reversed_path):
+                if branch not in self.managed_branches:
+                    if index == 0:
+                        self.add(
+                            branch=branch,
+                            opt_as_root=True,
+                            opt_onto=None,
+                            opt_yes=True)
+                    else:
+                        self.add(
+                            branch=branch,
+                            opt_onto=reversed_path[index - 1],
+                            opt_as_root=False,
+                            opt_yes=True)
+            print(fmt(f"Pull request `#{pr.number}` checked out at local branch `{pr.head}`"))
 
         debug('checkout_github_pr()',
               'Current GitHub user is ' + (current_user or '<none>'))
         self.__sync_annotations_to_definition_file(all_open_prs, current_user)
-
-        self.__git.checkout(LocalBranchShortName.of(pr.head))
-        print(fmt(f"Pull request `#{pr.number}` checked out at local branch `{pr.head}`"))
+        if pr:
+            self.__git.checkout(LocalBranchShortName.of(pr.head))
+            print(fmt(f"Switched to local branch `{pr.head}`"))
```

Comment:
Actually! one small glitch, on this very repo:

```
$ git machete github checkout-prs --all
Fetching origin...
Pull request #354 checked out at local branch feature/checkout_multiple_prs
Switched to local branch feature/checkout_multiple_prs
```

IMO it doesn't make sense to switch branch to any PR when `--all` has been passed... intuitively, I'd even say when any such option (`--mine`, `--by`, more than one number)... but leaving that to you to figure out (or to someone who follows after you, in case you get completely pulled away by Sherlock from now on ;) )

Reply:
A loose idea... maybe for clarity, `checkout-prs` could be left as is, and all the features of checking out multiple PRs could go into a new `reflect-PRs` subcommand? which does NOT switch any branch at the end. Although not sure if such complication would improve or rather worsen the UX

Reply:
Okay... so in other words a PR would be checked out if either:
1. there is exactly one PR in the repository, and it qualifies under `--all`, `--mine` or `--by=...` flag that has been provided by the user OR
2.  none of `--all`/`--mine`/`--by=...` has been provided; instead, a single PR number has been provided

Note that in the (2.) case, more than 1 entry might be added to .git/machete! since `checkout-prs`, even as of the current `master`, recursively adds the PRs for base branch

E.g. for https://github.com/VirtusLab/gpki/pull/63, I've run `git machete github checkout-prs 63` and got the machete status like:

![image](https://user-images.githubusercontent.com/3383210/151414030-730d69f6-76c6-4f12-8b94-a7e943f94ef3.png)

Reply:
Not really what i had in mind for (1.). I propose that we checkout PR when:

1. there is exactly one PR in the repository that qualifies under --all, --mine or --by=... flag that has been provided by the user OR (so there might be more PRs in the repo but only one qualifies)
2. none of --all/--mine/--by=... has been provided; instead, a single PR number has been provided

once we have that one PR from (1.) or (2.), we perform checks for the situation mentioned above ( parent branch is not a base branch) and if so, we checkout the most downstream branch that has not been merged. In this case it would be` docs/html_export` (interesting corner case: when `docs/html_export` has been merged to `enhance/master_or_main`, then we checkout the latter branch). But if that isn't logical enough (and it also looks pretty complicated) i would go for the option, where we don't checkout any branch in this situation.

Reply:
Actually, since the original idea is getting more complicated and less intuitive for the user, your idea to move all the features of checking out multiple PRs into a new subcommand, where we would only fetch them without checkouts and leaving `checkout-prs` as before, sounds much better now. I would go with the new idea.

Reply:
Okay! let's follow this path and see where it leads 🛣️

Reply:
> But if that isn't logical enough (and it also looks pretty complicated)

I actually have the experience of first figuring out and then using quite complex checkout rules for `git machete traverse`... i.e. which branch is checked out after the entire traversal is done. And even though the rules sounded logical initially... I often have a feeling myself that I don't know where I'm gonna end up after the traversal and why there 😅 so maybe better not to repeat this error here for `checkout-prs`